torchtitan/experiments/kernels/moe/indices.py

```diff
@@ -105,60 +147,68 @@ def generate_permute_indices(
     alignment: int,
     use_cpu: bool = False,
 ):
-    # Prepare permutation indices and the number of tokens for each expert.  The
-    # permutation indices are the indices of the tokens for each expert.  The
-    # number of tokens for each expert is the sum of the number of tokens for
-    # such experts from all ranks. This number is aligned to the provided
-    # alignment requirement (usually comes from group gemm).
-
-    # Args:
-    #     tokens_per_expert_group: number of tokens for each expert from all ranks.
-    #     experts_per_rank: number of experts per rank.
-    #     num_ranks: number of ranks.
-    #     max_len: maximum length of the output index vector. If greater than
-    #     total number of tokens, the remaining indices are set to -1.
-    #     alignment: alignment for each returned element in `m_sizes`.
-    #     use_cpu: whether to use cpu or gpu.
-    # Returns:
-    #     permuted_indices: permutation indices.
-    #     m_sizes: number of tokens for each expert.
-
-    # `tokens_per_expert_group` is of shape (num_ranks * experts_per_rank,), for example:
-    # From: |       rank 0      |       rank 1      |
-    # To:   | E0 | E1 | E2 | E3 | E0 | E1 | E2 | E3 |
-    #       |  4 |  2 |  1 |  3 |  1 |  2 |  3 |  4 |
-
```

Comment:
nit: can you preserve this illustration? thanks!

Reply:
yes, happy to.

Reply:
added, and also added the m_offsets to the Returns: doc string.